theano/gof/lazylinker_c.c

```diff
@@ -833,22 +854,27 @@ CLazyLinker_call(PyObject *_self, PyObject *args, PyObject *kwds)
             }
         }
 
-      for (int i = 0; i < self->n_output_vars && (!err); ++i)
-        {
-          err = lazy_rec_eval(self, self->output_vars[i], one, zero);
-        }
+        for (int i = 0; i < self->n_output_vars && (!err); ++i)
+          {
+            if (output_subset == NULL || output_subset[i] == 1)
+              {
+                err = lazy_rec_eval(self, self->output_vars[i], one, zero);
+              }
+          }
 
       if (!err)
         {
           // save references to outputs prior to updating storage containers
           assert (self->n_output_vars >= self->n_updates);
+          assert (output_subset_size < 0 || output_subset_size >= self->n_updates);
```

Comment:
The second part of the condition is only needed if the user can only skip the real output of the function and not the updates. Is that the case? I find that strange, as there is no code that mark the updates as needed to be computed in that PR.